### PR TITLE
gatk4_asereadcounter add updated meta and nf-tests

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/gatk4/asereadcounter/meta.yml [hunk 1]
```diff
@@ -25,18 +25,46 @@ input:
...
       pattern: "*.{bam,cram,sam}"
+  - input_index:
+      type: file
+      description: index file for input file
+      pattern: "*.{bai,crai}"
+  - meta2:
+      type: map
+      description: |
+        Groovy Map containing reference information
+        e.g. [ id:'test', single_end:false ]
```
Comment (mashehu) [inline modules/nf-core/gatk4/asereadcounter/meta.yml:36]:
```suggestion
        e.g. `[ id:'test', single_end:false ]`
```

File: modules/nf-core/gatk4/asereadcounter/meta.yml [hunk 2]
```diff
@@ -25,18 +25,46 @@ input:
...
       pattern: "*.{vcf.gz}"
+  - tbi:
+      type: file
+      description: index file for VCF file
+      pattern: "*.{vcf.gz.tbi}"
+  - meta3:
+      type: map
+      description: |
+        Groovy Map containing reference information
+        e.g. [ id:'reference' ]
```
Comment (mashehu) [inline modules/nf-core/gatk4/asereadcounter/meta.yml:49]:
```suggestion
        e.g. `[ id:'reference' ]`
```

File: modules/nf-core/gatk4/asereadcounter/meta.yml (hunk 1 above)
Comment (mashehu) [reply modules/nf-core/gatk4/asereadcounter/meta.yml:36]: is really `single_end` info about the reference?

File: modules/nf-core/gatk4/asereadcounter/tests/nextflow.config
Comment (mashehu) [inline modules/nf-core/gatk4/asereadcounter/tests/nextflow.config:1]: You can remove this file.

If the old pytests files are still there (including the entry in the tags.yml) please remove them as well

Comment (Lucpen) [reply modules/nf-core/gatk4/asereadcounter/tests/nextflow.config:1]: Done!

**Review by mashehu — APPROVED** [review]
Sorry, lost track of this one in the hackathon hubbub. looking good now!

**Comment by Lucpen** [comment]
Np, thanks!

**Pull request merged**